lib/statsd.js

```diff
@@ -464,5 +462,37 @@ Client.prototype.childClient = function(options) {
   return new ChildClient(this, options);
 };
 
+// Overrides tags in parent with tags from child with the same name (case sensitive) and return the result as new
+// array. parent and child are not mutated.
+function overrideTags (parent, child) {
+  var childCopy = {};
+  var toAppend = [];
+  child.forEach(function (tag) {
+    var idx = tag.indexOf(':');
```

Comment:
I think we're on close to the same idea here but aren't quite finding the common path.  :)  I agree that having no errors would not be the best solution- I just really, really don't want uncaught exceptions to be the norm, just based on my own experience with Node projects in production.  So all I ask is that the error is returned in a callback (or in the fallback error handler) instead of being an uncaught exception.  Does that work for you?